src/Enjin.Platform.Sdk/Enjin.Platform.Sdk/Json/GraphQlParameterJsonConverter.cs

```diff
@@ -32,8 +32,31 @@ public override void Write(Utf8JsonWriter writer, TValue value, JsonSerializerOp
 
         foreach (KeyValuePair<string, object?> variable in value.Parameters)
         {
-            writer.WritePropertyName(variable.Key);
-            writer.WriteRawValue(JsonSerializer.Serialize(variable.Value, options));
+          if (variable.Value != null && variable.Value.GetType().IsGenericType && variable.Value.GetType().GetGenericTypeDefinition() == typeof(List<>)) 
+          {
+                var arrayOfParams = (List<IGraphQlParameter>) variable.Value;
+                writer.WritePropertyName(variable.Key);
+                
+                foreach (var graphQlParameter in arrayOfParams)
+                {
+                    writer.WriteStartArray();
+                    writer.WriteStartObject();
+                    
+                    foreach (KeyValuePair<string, object?> parameter in graphQlParameter.Parameters)
```

Comment:
This probably doesn't work if there are nested arrays. I don't know if that ever happens in Enjin schemas though so maybe that is fine.

Reply:
You are right, I will make a recursive function though I wonder if there isn't another way, I mean this whole function is called recursively already and it goes inside each the objects and the objects of ojects and decodes them fine. Only for arrays it seems to fail, so I wonder if this can't be fixed in another way, I will give another thought tomorrow.